addons/mail/models/mail_activity.py

```diff
@@ -128,12 +134,12 @@ def _compute_res_name(self):
             activity.res_name = activity.res_model and \
                 self.env[activity.res_model].browse(activity.res_id).display_name
 
-    @api.depends('date_deadline')
+    @api.depends('active', 'date_deadline')
     def _compute_state(self):
         for record in self.filtered(lambda activity: activity.date_deadline):
             tz = record.user_id.sudo().tz
             date_deadline = record.date_deadline
-            record.state = self._compute_state_from_date(date_deadline, tz)
+            record.state = 'done' if not record.active else self._compute_state_from_date(date_deadline, tz)
```

Comment:
> the compute automatically set the state to “done” when archiving

but if there's no deadline but it's archived, it won't be marked as done no ?

because of that line
```py
for record in self.filtered(lambda activity: activity.date_deadline):